### PR TITLE
feat: [PagoPa-1155] add nav

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- added NAV (Numero Avviso) in the response

#### Motivation and Context

- new business request

#### How Has This Been Tested?

manual and junit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by alessio-acitelli** [comment]
update_code

File: src/main/java/it/gov/pagopa/debtposition/controller/pd/crud/api/impl/DebtPositionController.java [hunk 1]
```diff
@@ -56,7 +61,10 @@ public ResponseEntity<PaymentPositionModel> createDebtPosition(String organizati
         PaymentPosition createdDebtPos = paymentPositionService.create(debtPosition, organizationFiscalCode, toPublish);
 
         if (null != createdDebtPos) {
-            return new ResponseEntity<>(modelMapper.map(createdDebtPos, PaymentPositionModel.class), HttpStatus.CREATED);
+        	PaymentPositionModel paymentPosition = modelMapper.map(createdDebtPos, PaymentPositionModel.class);
+        	//PAGOPA-1155: add nav info to PO
+        	paymentPosition.getPaymentOption().forEach(po -> po.setNav(auxDigit+po.getIuv()));
```
Comment (andrea-deri) [inline src/main/java/it/gov/pagopa/debtposition/controller/pd/crud/api/impl/DebtPositionController.java:66]: Can the operation "auxDigit+po.getIuv()" (and so the injection of the auxDigit value from env parameter with @Value annotation) be included [in this class](https://github.com/pagopa/pagopa-debt-position/blob/main/src/main/java/it/gov/pagopa/debtposition/util/CommonUtil.java) as utility method? Surely can avoid code duplication and make code more clean.

**Review by andrea-deri — CHANGES_REQUESTED** [review]
Very good work! If you think the comment is useful, refactor the _auxDigit+iuv_ operation.

File: src/main/java/it/gov/pagopa/debtposition/controller/pd/actions/api/impl/DebtPositionActionsController.java [hunk 2]
```diff
@@ -24,14 +25,20 @@ public class DebtPositionActionsController implements IDebtPositionActionsContro
...
     @Override
     public ResponseEntity<PaymentPositionModel> publishDebtPosition(String organizationFiscalCode, String iupd) {
         log.info(String.format(LOG_BASE_HEADER_INFO, "POST", "publishDebtPosition", String.format(LOG_BASE_PARAMS_DETAIL, organizationFiscalCode, iupd)));
 
         PaymentPosition publishedDebtPos = paymentPositionActionsService.publish(organizationFiscalCode, iupd);
         if (null != publishedDebtPos) {
-            return new ResponseEntity<>(modelMapper.map(publishedDebtPos, PaymentPositionModel.class), HttpStatus.OK);
+        	PaymentPositionModel paymentPosition = modelMapper.map(publishedDebtPos, PaymentPositionModel.class);
+        	//PAGOPA-1155: add nav info to PO
+        	paymentPosition.getPaymentOption().forEach(po -> po.setNav(auxDigit+po.getIuv()));
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/debtposition/controller/pd/actions/api/impl/DebtPositionActionsController.java:40]: why not use `modelMapper` to set the nav?

**Review by jacopocarlini — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/debtposition/controller/pd/crud/api/impl/DebtPositionController.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/debtposition/controller/pd/crud/api/impl/DebtPositionController.java:66]: from a comparison with @jacopocarlini it was decided to leave the current enhancement of the NAV

File: src/main/java/it/gov/pagopa/debtposition/controller/pd/actions/api/impl/DebtPositionActionsController.java (hunk 2 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/debtposition/controller/pd/actions/api/impl/DebtPositionActionsController.java:40]: **modelmapper** is not a spring class. The **@value** annotation cannot be used

**Review by jacopocarlini — APPROVED** [review]
Good Job! 💪

**Pull request merged**